### PR TITLE
Fix WINDUP-742 Links to transformed files attached to classification …

### PR DESCRIPTION
**Pull request by OndraZizka**
…types instead of different files.

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/404/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1032/
Test PASSed.

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/jboss/GenerateJBossEjbDescriptorRuleProvider.java [hunk 1]
```diff
@@ -119,23 +118,19 @@ private void createReport(EvaluationContext evaluationContext, GraphContext cont
...
             for (VendorSpecificationExtensionModel vendorSpecificExtension : vendorSpecificService.getVendorSpecificationExtensions(ejbDescriptor))
             {
                 LOG.info("Vendor specific: " + vendorSpecificExtension.getFileName());
                 ClassificationModel classification = classificationService.attachClassification(evaluationContext, vendorSpecificExtension,
                             "EJB Specification Extension", "Vendor Specific EJB Specification Extension");
-                classificationService.attachLink(classification, generatedDescriptor);
             }
+
+            // Link up with the generated XML.
+            ejbDescriptor.addLinkToTransformedFile(newDescriptorLink);
```
Comment (mbriskar) [inline rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/jboss/GenerateJBossEjbDescriptorRuleProvider.java]: will this show in the final report?

**Comment by jsight** [comment]
Links do not appear in the report after this change.

**Comment by OndraZizka** [comment]
Could you please give me an example of a link which doesn't appear? I can see the links in the reports, at least in those I was checking when coding this.

**Comment by OndraZizka** [comment]
(00:21:46) ozizka-FN: The links actually should not be there
(00:22:09) ozizka-FN: It's part of the fix

**Comment by OndraZizka** [comment]
I have checked the rules. The links "missing" are actually links to files generated for other file, not for weblogic-ejb-jar.xml - that's what I think, I may be wrong, but consider the following log from the original run:

21:43:39,632 INFO  [org.jboss.windup.bootstrap.ConsoleProgressMonitor] (main) [53/319] MigrationRulesPhase - Generate jboss-ejb3.xml - Generate jboss-ejb3.xml_1
21:43:39,635 INFO  [org.jboss.windup.config.RuleSubset] (main) Rule [Generate jboss-ejb3.xml from org.jboss.windup.rules.apps.javaee.rules.jboss.GenerateJBossEjbDescriptorRuleProvider loaded from ModuleClassLoader for Module "org.jboss.windup.rules.apps.windup-rules-java-ee:2.4.0-SNAPSHOT_7b249ec0-2c27-47fc-8ed4-25f69d194193" from AddonModuleLoader] matched and will be performed.
21:43:39,663 INFO  [GenerateJBossEjbDescriptorRuleProvider] (main) Generated jboss-ejb3.xml for /home/ondra/work/Migration/Windup/test-files/jee-example-app-1.0.0.ear.report/archives/jee-example-services.jar/META-INF/ejb-jar.xml at: jboss_ejb3_JEE_Example_App__org_windup_example_jee_example_app_1_0_0_.xml
21:43:39,666 INFO  [GenerateJBossEjbDescriptorRuleProvider] (main) Vendor specific: weblogic-ejb-jar.xml
21:43:39,670 INFO  [GenerateJBossEjbDescriptorRuleProvider] (main) Generated jboss-ejb3.xml for /home/ondra/work/Migration/Windup/test-files/jee-example-app-1.0.0.ear.report/archives/jee-example-services2.jar/META-INF/ejb-jar.xml at: jboss_ejb3_JEE_Example_App__org_windup_example_jee_example_app_1_0_0_.1.xml
21:43:39,670 INFO  [GenerateJBossEjbDescriptorRuleProvider] (main) Vendor specific: weblogic-ejb-jar.xml
21:43:39,679 INFO  [org.jboss.windup.bootstrap.ConsoleProgressMonitor] (main) [54/319] MigrationRulesPhase - Generate jboss-web.xml - Generate jboss-web.xml_1
21:43:39,681 INFO  [org.jboss.windup.config.RuleSubset] (main) Rule [Generate jboss-web.xml from org.jboss.windup.rules.apps.javaee.rules.jboss.GenerateJBossWebDescriptorRuleProvider loaded from ModuleClassLoader for Module "org.jboss.windup.rules.apps.windup-rules-java-ee:2.4.0-SNAPSHOT_7b249ec0-2c27-47fc-8ed4-25f69d194193" from AddonModuleLoader] matched and will be performed.
21:43:39,697 INFO  [GenerateJBossWebDescriptorRuleProvider] (main) Generated jboss-web.xml for /home/ondra/work/Migration/Windup/test-files/jee-example-app-1.0.0.ear.report/archives/jee-example-web.war/WEB-INF/web.xml at: jboss_web_JEE_Example_App__org_windup_example_jee_example_app_1_0_0_.xml
21:43:39,702 INFO  [GenerateJBossWebDescriptorRuleProvider] (main) Vendor specific: weblogic.xml

This shows that the two files are generated for other source files and they are linked only because of shared classification, what is the bug being fixed by this.

As I say, I may be wrong. Please check.

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/jboss/GenerateJBossEjbDescriptorRuleProvider.java (hunk 1 above)
Comment (OndraZizka) [inline rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/jboss/GenerateJBossEjbDescriptorRuleProvider.java]: Yes.

**Comment by OndraZizka** [comment]
Correction: The links were not appearing where they should. However in weblogic-ejb-jar.xml they won't appear as we have no generated file for it.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1040/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/412/
Test PASSed.

**Comment by jsight** [comment]
This looks good, although I don't think it goes quite far enough. I'll tweak it slightly and merge soon.

Thanks @OndraZizka!

**Pull request merged**